docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'cent…

kollmana opened the pull request:
…ral aspects of security'

based on new block 'Security'

Blaupausen - Referenzarchitektur - 4. Die Software-Referenzarchitektur: 
neues Unterkapitel 'Zentrale Sicherheitsaspekte in der Referenzarchitektur'
- Verpflichtende Einbindung von Querschnitt-Baustein 'Security' für Single Sign On (SSO) zur Authentifizierung und Autorisierung
- Protokolle: OAuth2.0 und OpenIDConnect  mit Verweis auf Kapitel in Baustein 'Security'
- AWL-Diagramm: Anfragen über Portal, Service-Gateway, von Batch/Task gehen an  Baustein 'Security' u. weiter an IAM
- Anfragen über Portal, Service-Gateway, von Batch/Task mit Verweis auf Kapitel in Baustein 'Security'

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

based on new block 'Security'

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

based on new block 'Security'

kollmana committed:
Merge branch 'master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl.dn.svg:
Den gelben Kasten mit dem Baustein "Security" würde ich entfernen, da es kein eigenes System ist, das sich dort befindet. Im gelben Kasten zu IAM bitte nur "IAM-Service" reinschreiben.

Stattdessen würde ich rechts neben das Service-Gateway auch noch das Authentication Gateway einzeichnen (siehe isy-security).

kollmana replied:
Im Text vorher wird auf den Querschnittsbaustein 'Security' Bezug genommen, das IAM-System dagegen nicht erwähnt. Vielleicht sollte besser sowohl im Text als auch im Diagramm von 'Autorisierung' gesprochen werden - wie im Diagramm vorher (Figure 11. Aufrufbeziehungen für eine Geschäftsanwendung mit GUI und XML-Schnittstelle).

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+Die IT-Sicherheit ist insbesondere dann für Softwaresysteme eine wichtige Anforderung, wenn diese sensible oder gar personenbezogene Daten verarbeiten.
+Solche Systeme stellen hohe Anforderungen an Vertraulichkeit, Integrität und Verfügbarkeit.
+Eine Referenzarchitektur muss als grundlegendes Element der IT-Sicherheitsstrategie die Einhaltung solcher Schutzziele unterstützen.
```
Diese drei Zeilen würde ich entfernen, da IT-Sicherheit immer wichtig ist.

kollmana replied:
War als Einleitung und Rechtfertigung für den Baustein 'Security' gedacht - kann aber weg.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Solche Systeme stellen hohe Anforderungen an Vertraulichkeit, Integrität und Verfügbarkeit.
+Eine Referenzarchitektur muss als grundlegendes Element der IT-Sicherheitsstrategie die Einhaltung solcher Schutzziele unterstützen.
+
+Ein Pfeiler der IT-Sicherheitsstrategie ist der Schutz der Daten und IT-Systeme vor unberechtigter Nutzung.
```
Welche IT-Sicherheitsstrategie? Vom BVA? Wenn nicht referenzierbar, dann bitte die Zeile entfernen.

kollmana replied:
BVA: http://gsb.download.bva.bund.de/BIT/V-Modell_XT_Bund/V-Modell%20XT%20Bund%20HTML/f436f8cfc083ae.html#toc472

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+Ein Pfeiler der IT-Sicherheitsstrategie ist der Schutz der Daten und IT-Systeme vor unberechtigter Nutzung.
+Die IsyFact Referenzarchitektur legt fest, dass jegliche Nutzung von IsyFact-Anwendungen nur authentifizierten und dafür autorisierten Benutzern möglich ist.
+Diese Forderung wird im Querschnitt-Baustein Security umgesetzt, den alle IsyFact-Anwendungen zwingend bei der Kommunikation, sowohl mit anderen Anwendungen innerhalb der Anwendungslandschaft, als auch mit Systemen außerhalb und auch bei der Kommunikation mit Nutzern, einsetzen müssen.
```
Die Kommunikation zwischen Systemen ist an dieser Stelle erst einmal nicht relevant und würde ich entfernen. 
Die Bausteinbeschreibung würde ich auch weiter nach hinten stellen. Wichtig ist hier zunächst zu erläutern:
1.) Für die Authentifizierung wird ein zentraler Service (IAM-Service) in der Anwendungslandschaft genutzt. 
2.) Die Zugriffswegen sind entweder 1.) über das Portal für Anwender, 2a) über die Service-Gateways für externe Systeme, 2b) über das Authentication-Gateway für externe Systeme und 3) direkt am IAM-Service für interne Systeme einer Anwendungslandschaft. Diese Zugriffswegen müssen grob beschrieben werden (kein OAuth2 Spezifika) und dann zu den relevanten Dokumenten verlinkt werden.
3.) Die Autorisierung findet an jeder Systemgrenze statt. Je nach Schutzbedarf und Anforderung kann auch innerhalb einer IT-Systems eine zusätzliche Autorisierung stattfinden. 
4.) Für die Bausteinbeschreibung würde ich einfach nur in wenigen Sätzen erwähnen, dass die Funktionen zur Authentifizierung und Autorisierung durch den Baustein "Security" umgesetzt werden. Dies gilt nicht für die Authentifizierung am Portal, die durch das Portal umgesetzt wird (hier wird der Baustein isy-security nicht verwendet), sondern der Portal Apache greift direkt auf den IAM-Service zu.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Externe Anwendungen gelangen über ein Service-Gateway zu den Services der Geschäftsanwendung.
+Hier prüft das Service-Gateway bei jeder Anfrage, ob das anfragende System dazu berechtigt ist.
+Dazu muss sich die Anwendung gegenüber dem zentralen IAM-System über den xref:isyfact-standards-doku:isy-security:konzept/master.adoc#client-credential-flow[_Client Credential Flow_] mit seiner Client-ID u. -Secret authentifizieren.
+Alternativ kann auch das alte Verfahren xref:isyfact-standards-doku:isy-security:konzept/master.adoc#resource-owner-password-credential-flow[_Resource-Owner Password-Credentials Flow_] mit Anwendungskennung und Kennwort verwendet werden (siehe: xref:isyfact-standards-doku:isy-security:konzept/master.adoc#aufruf-service-gateway[Konzept 'Security': Aufruf Service-Gateway]).
```
Die einzelnen OAuth2 Spezifika müssen hier nicht genannt werden. Es reicht ein Satz, dass OAuth2 verwendet wird.

mayrklaus replied:
Für mich stellt sich die Frage, ob ein Angreifer das Service-Gateway oder das Portal umgehen kann. Liegt dann die Anwendung offen? Wie schützt sich die Anwendung für diesen Fall?

kollmana committed:
Merge remote-tracking branch 'origin/master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Changes from reviews

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Changes from reviews

kollmana committed:
Merge remote-tracking branch 'origin/master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

kollmana committed:
Merge branch 'master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl.dn.svg:
Es gibt zwei Möglichkeiten:
Wenn Authentication Gateway existiert, dann sollte es keinen Aufruf vom Service-Gateway auf das IAM geben
Wenn es nicht existiert, sollte man es in der Grafik weglassen.
Bitte klären welche Variante wir dokumentieren. ggf beide?

kollmana replied:
Das Auth-GW wird in der nächsten / einer zukünftigen Version von IsyFact eingeführt und sollte hier explizit aufgeführt werden. Im Text ist das 'zukünftig' auch erwähnt und dass die Authentifizierunung vom S-GW zum Auth-GW wandert.

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+[[zentrale-sicherheitsaspekte]]
+=== Zentrale Sicherheitsaspekte in der Referenzarchitektur
+
+Die IsyFact Referenzarchitektur legt fest, dass jegliche Nutzung von IsyFact-Anwendungen nur authentifizierten und dafür autorisierten Benutzern möglich ist.
```
Bitte vermeiden, dass synonyme Begriffe erscheinen Anwender / Benuter

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+Von Außen können IsyFact-Anwendungen und deren Services nur über Zugangsservices in der Informations- u. Dienstezone erreicht werden:
+
+- Interne und externe menschliche Anwender, die die Anwendung über deren Web-Oberfläche (im Web-Browser) bedienen, werden über das Portal geleitet.
```
menschlich ist hier überflüssig

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+- Interne und externe menschliche Anwender, die die Anwendung über deren Web-Oberfläche (im Web-Browser) bedienen, werden über das Portal geleitet.
+
+- Externe Systeme werden über ein Service-Gateway geleitet.
+In zukünftigen Versionen von IsyFact muss das externe System erst über das Authentication-Gateway die Berechtigung zur Nutzung einholen.
```
von wem und wie?

mayrklaus replied:
und was ist in der jetzigen Version?

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+Die Zugangsberechtigung wird dabei in zwei Schritten überprüft:
+
+. Die Zugangsservices prüfen die Identität des Aufrufers und lassen nur für die Anwendungslandschaft zugelassene durch.
```
was sind Zugangsservices? Der Begrif ist hier (und weiter oben) erst man noch undefiniert

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,45 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+. Die Zugangsservices prüfen die Identität des Aufrufers und lassen nur für die Anwendungslandschaft zugelassene durch.
+Für die Authentifizierung wird ein zentraler Service (IAM-Service) in der Anwendungslandschaft genutzt.
+Der IAM-Service verifiziert die Identität des Anwenders bzw. Systems anhand eindeutiger Eigenschaften (z.B. biometrische Daten) oder geheimer Daten (z.B. Passwort).
+Die Identität und die damit verknüpften Rollen in der Anwendungslandschaft werden in der Anwenderverwaltung gepflegt.
```
Wenn man dies erwähnt, dann ist man auf einer technischen Ebene, in der der Leser dann wirklich auch die Details wissen will: was passiert mit den gepflegten Berechtigungen? Wo werden sie übermittelt, wie werden sie ausgewertet?
und wie wird sichergestellt dass diese Informationen nicht verfälscht werden

mayrklaus reviewed (commented):
Es wäre wahrsche3inlich besser die technischen Details (Ablauf inklusive Prüfungen) nur im Baustein zu beschreiben 

Besser wäre wenn wir hier kurz nur die im Bild gezeigten Komponenten 
im Text nochmal kurz auflisten würden und dabei dazuschreiben, welche Aufgaben diese 
Komponenten haben, und mit welche Daten sie arbeiten

vielleicht zwei Bilder für alte und neue Architektur.
und generell die Szenarien (technischer User oder menschlicher User greift zu) benennenen

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Rewrite from reviews

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Rewrite from reviews

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Client certificate as additional factor,
diagram simplified

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Client certificate as additional factor,
diagram simplified

kollmana committed:
Merge branch 'master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,37 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+- Ein vorgeschalteter Loadbalancer (nicht im Bild) verlangt ein Client-Zertifikat des BVA/ITZ-Bund.
+Clients (Browser oder Anwendung), die nicht im Besitz dieses Client-Zertifikats sind, werden direkt abgewiesen.
+-  Im Portal bzw. Gateway wird die Identität des Nutzers geprüft (Authentifizierung).
+Der Nutzer muss dafür geheimes Wissen übermitteln (Name und Passwort für Anwender bzw. Client-ID und Secret für Anwendungen), das an einen zentralen Service (IAM-Service) in der Anwendungslandschaft weitergereicht wird.
```
nur Name und Passwort? Zusätzlich könnte hier auch noch ein 2FA Token verlangt werden

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,37 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Ist der Nutzer dem IAM-Service bekannt, stellt er ein Access-Token aus, das auch die dem Nutzer zugeordneten Rollen enthält.
+
+Der Aufruf des Nutzers wird zusammen mit diesem Access-Token an die (Interaktions- bzw. Service-Schicht /) Systemgrenze der IsyFact-Fachanwendung weitergeleitet.
+Dort wird (über ein Rolle-Rechte Mapping) geprüft, ob der Nutzer das zur Ausführung des gewünschten Service notwendige Recht besitzt (Autorisierung).
```
Hier könnte man noch erklären, dass die mitgelieferten Berechtigungen aus dem AccessToken entnommen 
und mit den geforderten Berechtigungen abgeglichen werden

kollmana replied:
Allgemeiner: Authentifizierungsdaten des Nutzers

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl-2.dn.svg:
Mit dem Bild bin ich (nach wie vor) nicht ganz einverstanden
* wenn wir in der neuen Architektur sind, dann redet das Service-Gateway nicht mit dem IAM
* wenn wir in der alten Architektur sind, dann gibt es dort kein Authentication GW.

=> entweder beide Architekturen erläutern (mit zwei Bildern) oder nur die neue Architektur skizzieren

kollmana replied:
Neue Architektur (S-GW + Auth-GW) u. Text: Aktuell macht S-GW auch Auth.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Future Architecture with Auth-GW as fact, S-GW currently doing its job.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Future Architecture with Auth-GW as fact, S-GW currently doing its job.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Text refinement

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Text refinement

stdll commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,38 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Bevor der Zugang zu der IsyFact-Anwendung gewährt wird, erfolgt in der Informations- und Dienstezone eine Zweifaktor-Berechtigungsprüfung:
+
+- Ein vorgeschalteter Loadbalancer (nicht im Bild) verlangt ein Client-Zertifikat des BVA/ITZ-Bund.
+Clients (Browser oder Anwendung), die nicht im Besitz dieses Client-Zertifikats sind, werden direkt abgewiesen.
```
Diese Zeilen sind BVA-spezifisch und gelten ggf. nicht für andere Nutzer der IsyFact. Vielleicht kann man den Teil  als Empfehlung formulieren und konkrete Behördennamen vermeiden.

stdll commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl.dn.svg:
Wird das Bild im Text verwendet? Falls nicht, bitte entfernen.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Client certificate is optional

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Client certificate is optional

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -485,21 +485,22 @@ Aktuell wird diese Funktion vom Service-Gateway mit übernommen.
 
 NOTE: IAM steht für xref:glossary:literaturextern:inhalt.adoc#litextern-identity_management[Identity and Access Management].
 
+Weitere Sicherheit bietet die zusätzliche Verwendung eines Client-Zertifikats:
```
Wenn der Loadbalancer dieses Zertifikat prüft, dann stellt sich die Frage, welche Komponente im Load-Balancer diese Aufgabe übernimmt. Muss dazu die security-Komponente auf den Loadbalancer deployed werden?
Ich glaube dass das eher nicht der Fall ist. 
Somit beschreibst du hier in deinem neuen Kapitel eine Funktionalität, die außerhalb der Funktionalität der Security-Komponente liegt. Wenn wir diese Funktionalität hier wirklich beschreiben wollen, dann sollten wir dies in einem anderen Kapitel machen.

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -485,21 +485,22 @@ Aktuell wird diese Funktion vom Service-Gateway mit übernommen.
 NOTE: IAM steht für xref:glossary:literaturextern:inhalt.adoc#litextern-identity_management[Identity and Access Management].
 
+Weitere Sicherheit bietet die zusätzliche Verwendung eines Client-Zertifikats:
+Die betreibende Behörde stellt ein Client-Zertifikat aus und verteilt es an alle berechtigten Nutzer.
```
Prinzipiell ist nicht wichtig ob der Aussteller des Client-Zertifikats eine Behörde ist.
SW und Doku die wir unter GitHub bereitstellen haben m.E. den Anspruch auch außerhalb eines Behörden-Umfeldes zu laufen

dauna-msg replied:
Ja, ich würde die Information auch weglassen, wer das Zertifikat ausstellt.

kollmana replied:
Rausgenommen

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,39 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Sicherheit ist ein Ziel der Referenzarchitektur (<<die-ziele-der-referenzarchitektur>>)
+und technische Querschnittsaufgabe.
+Ein Aspekt davon ist der Schutz vor unberechtigtem Zugriff auf vertrauliche Daten.
+Wie bereits in <<image-GA-AW2>> angedeutet, können Nutzer einer IsyFact-Anwendung nicht direkt darauf zugreifen, sondern nur über eine dazwischenliegenden Informations- und Dienstezone:
```
Mir fehlt hier noch ganz prägnant: 
1.) Die Authentifizierung von Anwendern findet an den Außengrenzen der Anwendungslandschaft statt. Zugriff auf die Anwendungslandschaft bieten Portalanwendungen, Service-Gateways und zukünftig das Authentication-Gateway.
Jeder Zugriff innerhalb der Anwendungslandschaft ist somit authentifiziert. 
Eine Autorisierung von Zugriffen sollte ebenfalls an den Außengrenzen geschehen.

2.) Die Autorisierung von Zugriffen muss an den Systemgrenzen der Anwendungen stattfinden. Es dürfen nur authentifizierte und autorisierte Zugriffe auf eine Anwendung möglich sein.

3.) Die Authentifizierung erfolgt über einen zentralen Querschnittsservice je Anwendungslandschaft, dem IAM-Service.

kollmana replied:
1.)
An den Außengrenzen der Anwendungslandschaft erfolgt die Authentifizierung des Anwenders und an den Systemgrenzen der Anwendungen die Prüfung der Autorisierung zur Nutzung des angefragten Zugriffs.

Geschäftsanwendungen einer IsyFact Anwendungslandschaft sind für Anwender von außen nur über die Zugriff-steuernden Komponenten Portalanwendung, Service-Gateway und zukünftige Authentication-Gateway erreichbar, die somit die sichtbare Außengrenze der Anwendungslandschaft darstellen.
Dort erfolgt die Authentifizierung des Anwenders (_schwarzes Schlüssel_-Symbol) mit Hilfe eines zentralen Querschnittservice je Anwendungslandschaft, dem IAM-Service.
Jeder Zugriff innerhalb der Anwendungslandschaft ist somit authentifiziert (_grünes Schlüssel_-Symbol).
Eine grobe Autorisierung von Zugriffen sollte ebenfalls an den Außengrenzen geschehen, z.B. durch Client-Zertifikate.

2.)
Die feingranulare Autorisierung zur Verwendung einer Funktion oder eines Service einer IsyFact-Geschäftsanwendung muss an den Systemgrenzen der Anwendung stattfinden (_Schloss_-Symbol). Es dürfen nur authentifizierte und autorisierte Zugriffe auf eine Anwendung möglich sein (_Dokument mit grünem Schlüssel_-Symbol).

3.)
Dort [Außengrenze] erfolgt die Authentifizierung des Anwenders (_schwarzes Schlüssel_-Symbol) mit Hilfe eines zentralen Querschnittservice je Anwendungslandschaft, dem IAM-Service.

Die Authentifizierungs- und Autorisierungsprotokolle _OpenID Connect_ und _OAuth2.0_ definieren verschiedene Verfahren zur Authentifizierung der Anwenderanmeldedaten (_schwarzes Schlüssel_-Symbol)  gegenüber dem IAM und die Ausstellung eines Access Token (_grünes Schlüssel_-Symbol) zur Autorisierung an den Systemgrenzen der Anwendung.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl.dn.svg:
Ich würde den gelben Kasten nur "IAM-Service" nennen. Dadrunter "Querschnitts-System" in "Querschnittsanwendung" umformulieren. 
Links die blauen Kästen würde ich jeweils "Geschäftsanwendung" benennen, analog zu https://isyfact.github.io/isyfact-standards-doku/current/blaupausen/referenzarchitektur/master.html#fachliche-referenzarchitektur.

Hilfreich ist es, wenn aus der Abbildung hervorgeht, wo die Authentifizierung startet, die Authentifizierung durchgeführt wird und die Autorisierung stattfindet. Man könnte dazu am Portal, Service-Gateway und Auth-Gateway z.B. ein kleines Schlüssel-Symbol oder ähnliches am eingehenden Request als Authentifizierungsstart einbringen. Am IAM-Service dann ein anderes Symbol für die Authentifizierung und z.B. ein Schloss-Symbol zur Darstellung der Autorisierung an den QAs und GAs, wo die Anfragen eintreffen.

Ich finde diese Abbildung aber auch besser als die obere, da die unterschiedlichen Zonen hier nicht relevant sind und das ganze dann eher in die techn. Schiene bringen. Wir sollten hier aus fachlicher Sicht einen Überblick geben.

kollmana replied:
Ausgehend von Abbildung 1: 
- Auth-GW hinzugefügt (neben Service-GW) 
- Abbildung nur der zukünftigen Architektur: ext.Anw.: Auth über Auth-GW; authentifizierte Zugriffe über Service-GW
- IAM-Service als explizite Querschnittsanwendung
- Anfragen u. Antworten mit Symbolen versehen: 
- - Authentifizierungsdaten des Anwenders (ext.Anwendung / Portal -> IAM)
- - Anfrage mit fachlichen Daten u. Authentifizierungsdaten (Web-Client -> Portal)
- - IAM: Identitäts- u. Schlüsselverwaltung (Authentifizierung, Ausstellung Access-Token)
- - AccessToken nach der Authentifizierung durch IAM-Service (Antwort IAM -> Portal / ext. Anwendung)
- - Anfrage mit fachlichen Daten u. Access-Token (ext.Anwendung / Portal -> GA)
- - abgesicherte Schnittstelle der Geschäftsanwendung (Endpunkt der Anfrage mit fachlichen Daten in der GA)
- Legende (s.o.)

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,39 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+Weitere Sicherheit bietet die zusätzliche Verwendung eines Client-Zertifikats:
+Die betreibende Behörde stellt ein Client-Zertifikat aus und verteilt es an alle berechtigten Nutzer.
+Ein dem Portal bzw. Authentication-Gateway vorgeschalteter Loadbalancer (nicht im Bild) verlangt und prüft dieses Client-Zertifikat.
```
Den Loadbalancer würde hier nicht erwähnen, sondern dass beim Betreten der Anwendungslandschaft das Zertifikat geprüft wird und Informationen aus dem Zertifikat, wie z.B. das Behördenkennzeichen, des Aufrufers zur Authentifizierung oder Autorisierung in der Anwendungslandschaft genutzt werden können. Kurze Erwähnung, dass der Baustein isy-security das aus dem Zertifikat extrahierte BHKNZ als zweiten Faktor in der Authentifizierung verwendet, sofern es verfügbar ist.

kollmana replied:
Eine grobe Autorisierung von Zugriffen sollte ebenfalls an den Außengrenzen geschehen, z.B. durch Client-
Zertifikate.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Included review remarks from A.Daun

kollmana committed:
Merge remote-tracking branch 'origin/feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

# Conflicts:
#	isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl.dn.svg
#	isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc

kollmana commented:
- Reworked Diagram: with symbols (key, lock) to show flow and functionality
- Reworked Text: more functianal, less technical

kollmana committed:
Merge branch 'master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Jeder Zugriff innerhalb der Anwendungslandschaft ist somit authentifiziert (grüner Schlüssel).
+Eine grobe Autorisierung von Zugriffen sollte ebenfalls an den Außengrenzen geschehen, z.B. durch Client-Zertifikate.
+
+Die feingranulare Autorisierung zur Verwendung einer Funktion oder eines Service einer IsyFact-Geschäftsanwendung muss an den Systemgrenzen der Anwendung stattfinden (Schloss an der externen Schnittstelle der Anwendung). Es dürfen nur authentifizierte und autorisierte Zugriffe auf eine Anwendung möglich sein (Dokument mit grünem Schlüssel).
```
Statt "Dokument mit grünem Schlüssel" (hier weiß man nicht, welches Dokument du meinst, bzw. welche Rolle die Farbe des Schlüssels hier spielt) würde ich schreiben; "Der Anwender hat passenden Schlüssel für das Schloss"

mayrklaus replied:
Jetzt ists klarer, du nimmst hier auf die Grafik Bezug, 
Ums noch deutlicher zu schreiben, könntest du sagen (grüner Schlüssel im Bild oben)
Das ist aber kein Muss

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+=== Zentrale Sicherheitsaspekte in der Referenzarchitektur
+
+Sicherheit ist ein Ziel der Referenzarchitektur (<<die-ziele-der-referenzarchitektur>>)
+und technische Querschnittsaufgabe.
```
```suggestion
Ziele der Referenzarchitektur (<<die-ziele-der-referenzarchitektur>>) ist die Gewährleistung von Sicherheitsaspekten und die Erfüllung technischer Querschnittsaufgaben.
```

kollmana replied:
**Ein** Ziel der Referenzarchitektur (<<die-ziele-der-referenzarchitektur>>) ist die Gewährleistung von Sicherheitsaspekten als technische Querschnittsaufgabe.

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+Sicherheit ist ein Ziel der Referenzarchitektur (<<die-ziele-der-referenzarchitektur>>)
+und technische Querschnittsaufgabe.
+Ein Aspekt davon ist der Schutz vor unberechtigtem Zugriff auf vertrauliche Daten.
```
```suggestion
Ein Aspekt zum Thema Sicherheit ist der Schutz vor unberechtigtem Zugriff auf vertrauliche Daten.
```

kollmana replied:
Ein Teilbereich davon {Sicherheitsaspekte} ...

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Sicherheit ist ein Ziel der Referenzarchitektur (<<die-ziele-der-referenzarchitektur>>)
+und technische Querschnittsaufgabe.
+Ein Aspekt davon ist der Schutz vor unberechtigtem Zugriff auf vertrauliche Daten.
+IsyFact gewährleistet das durch die Implementierung der Autorisierungs- und Authentifizierungsprotokolle OAuth2.0 und OpenID Connect.
```
```suggestion
IsyFact gewährleistet das durch die Implementierung der standardisierten Protokolle zur Authentifizierung (OpenID Connect) und Autorisierung (OAuth2.0).
```
Da im Prozessflow die Authentifizierung vor der Autorisierung kommt wäre mein Vorschlag dies auch im Text in dieser Reihenfolge zu halten.
Eventuell könnte man statt der konkreten Protokollnamen in Klammern auch einen jeweiligen externen Link hinter die Worte _Authentifizierung_ bzw. _Autorisierung_ legen und somit im lesbaren Text gänzlich frei von der Konkretisierung technischer Details sein.

kollmana replied:
Übernommen. 
In isy-security.inhalt und isy-documentation: literaturextern.inhalt wurden die links zu OAuth u. OIDC entfernt. 
Erklärt werden die Begriffe im Konzept 'Security' - OAuth2.0 und OpenIDConnect.

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Ein Aspekt davon ist der Schutz vor unberechtigtem Zugriff auf vertrauliche Daten.
+IsyFact gewährleistet das durch die Implementierung der Autorisierungs- und Authentifizierungsprotokolle OAuth2.0 und OpenID Connect.
+
+Anwender erhalten Zugriff auf Geschäftsanwendungen innerhalb der IsyFact Anwendungslandschaft nur über Portalanwendungen, Service-Gateways und zukünftig das Authentication-Gateway, die die Außengrenzen der Anwendungslandschaft darstellen.
```
```suggestion
Geschäftsanwendungen einer IsyFact Anwendungslandschaft sind für Anwender von außen nur über die Zugriff-steuernden Komponenten Portalanwendung, Service-Gateway und zukünftige Authentication-Gateway erreichbar, die somit die sichtbare Außengrenze der Anwendungslandschaft darstellen.
```

kollmana replied:
Übernommen. 
So (passiv) liegt der Fokus auf den Geschäftsanwendungen.

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+IsyFact gewährleistet das durch die Implementierung der Autorisierungs- und Authentifizierungsprotokolle OAuth2.0 und OpenID Connect.
+
+Anwender erhalten Zugriff auf Geschäftsanwendungen innerhalb der IsyFact Anwendungslandschaft nur über Portalanwendungen, Service-Gateways und zukünftig das Authentication-Gateway, die die Außengrenzen der Anwendungslandschaft darstellen.
+Dort erfolgt die Authentifizierung des Anwenders (schwarzer Schlüssel) mit Hilfe eines zentralen Querschnittservice je Anwendungslandschaft, dem IAM-Service.
```
```suggestion
Die folgende Beschreibung nimmt Bezug auf die nachfolgende Abbildung durch Erwähnung von:
- _schwarzes Schlüssel_-Symbol: Authentifizierungsdaten des Anwenders
- _grünes Schlüssel_-Symbol: gültiges AccessToken nach der Authentifizierung durch IAM-Service
- _Dokument_-Symbol: Anfrage mit fachlichen Daten
- _Schloss_-Symbol: per Rollen abgesicherte Schnittstelle der Geschäftsanwendung
- Kombinationen aus den vier zuvor genannten Symbolen

Alle fachlichen Anfragen von noch nicht angemeldeten Anwendern (_schwarzes Schlüssel_-Symbol) werden in diesen Zugriffs-Komponenten im Zusammenspiel mit dem IAM-Service als der zentrale Querschnittservice jeder Anwendungslandschaft authentifiziert.
```

kollmana replied:
Legende in Diagramm
Bezeichnungen in Text übernommen

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+Anwender erhalten Zugriff auf Geschäftsanwendungen innerhalb der IsyFact Anwendungslandschaft nur über Portalanwendungen, Service-Gateways und zukünftig das Authentication-Gateway, die die Außengrenzen der Anwendungslandschaft darstellen.
+Dort erfolgt die Authentifizierung des Anwenders (schwarzer Schlüssel) mit Hilfe eines zentralen Querschnittservice je Anwendungslandschaft, dem IAM-Service.
+Jeder Zugriff innerhalb der Anwendungslandschaft ist somit authentifiziert (grüner Schlüssel).
```
```suggestion
Jeder Zugriff innerhalb der Anwendungslandschaft ist somit authentifiziert (_grünes Schlüssel_-Symbol).
```

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+Jeder Zugriff innerhalb der Anwendungslandschaft ist somit authentifiziert (grüner Schlüssel).
+Eine grobe Autorisierung von Zugriffen sollte ebenfalls an den Außengrenzen geschehen, z.B. durch Client-Zertifikate.
+
+Die feingranulare Autorisierung zur Verwendung einer Funktion oder eines Service einer IsyFact-Geschäftsanwendung muss an den Systemgrenzen der Anwendung stattfinden (Schloss an der externen Schnittstelle der Anwendung). Es dürfen nur authentifizierte und autorisierte Zugriffe auf eine Anwendung möglich sein (Dokument mit grünem Schlüssel).
```
```suggestion
Die feingranulare Autorisierung beim Aufruf einer internen Geschäftsanwendung erfolgt an den Methoden oder dem Service der Anwendung selbst (_Schloss_-Symbol). Es dürfen nur authentifizierte und autorisierte Zugriffe auf eine Geschäftsanwendung möglich sein (_Dokument mit grünem Schlüssel_-Symbol).
```

kollmana replied:
Zugriff von Außen = Aufruf einer Anwendungsfunktion der Geschäftsanwendung = technisch: externe Schnittstelle. 
Diese externe Schnittstelle muss geschützt sein: 
- Authentifiziert: Prüfung des Access-Tokens an sich (Aussteller, Gültigkeit)
- Autorisiert: Wurde das erforderliche Recht (Secured-Annotation) gewährt (über Rollen-Rechte-Mapping durch die Rolle)?

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Textual description of icons after diagram

kollmana committed:
Merge branch 'master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+.Integration von IAM in eine Anwendungslandschaft
+[id="image-iam-integration-awl",reftext="{figure-caption} {counter:figures}"]
+image::referenzarchitektur/sicherheit_iam_integration_awl.dn.svg[]
```
**Dickes Lob! Das ist ein sehr gutes und detailreiches Diagramm...**

Mir wurde der Gehalt des Bildes erst durch den Text und eine längere Verarbeitung der Symbole klar.
Daher würde ich dem Bild eine Legende spendieren um die Leser abzuholen, die nicht den Text lesen.
Einen Vorschlag habe ich oben in der Anmerkung zu Zeile 480 gemacht.

kollmana replied:
Legende hinzugefügt

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+NOTE: IAM steht für xref:glossary:literaturextern:inhalt.adoc#litextern-identity_management[Identity and Access Management].
+
+Die Autorisierungs- und Authentifizierungsprotokolle OAuth2.0 und OpenID Connect definieren verschiedene Verfahren zur Authentifizierung gegenüber dem IAM und die Ausstellung eines Access Token (grüner Schlüssel) zur Autorisierung an den Systemgrenzen der Anwendung.
```
```suggestion
Die Authentifizierungs- und Autorisierungsprotokolle _OpenID Connect_ und _OAuth2.0_ definieren verschiedene Verfahren zur Authentifizierung der Anwenderanmeldedaten (_schwarzes Schlüssel_-Symbol) gegenüber dem IAM und die Ausstellung eines Access Token (_grüne Schlüssel_-Symbol) zur Autorisierung an den Systemgrenzen der Anwendung.
```

kollmana replied:
Übernommen

huebnercl commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,30 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+NOTE: IAM steht für xref:glossary:literaturextern:inhalt.adoc#litextern-identity_management[Identity and Access Management].
+
+Die Autorisierungs- und Authentifizierungsprotokolle OAuth2.0 und OpenID Connect definieren verschiedene Verfahren zur Authentifizierung gegenüber dem IAM und die Ausstellung eines Access Token (grüner Schlüssel) zur Autorisierung an den Systemgrenzen der Anwendung.
+Eine detaillierte Beschreibung der erwähnten Systeme (Portal, Service-Gateway, Authentication-Gateway, IAM), der implementierten Authentifizierungsverfahren und des Access-Tokens findet sich in xref:isyfact-standards-doku:isy-security:konzept/master.adoc#inhalt[Konzept des Querschnitt-Bausteins Security].
```
```suggestion
Eine detaillierte Beschreibung der erwähnten Zugriffskomponenten (Portal, Service-Gateway, Authentication-Gateway),  des IAM-Service, der implementierten Authentifizierungsverfahren und des Access-Tokens findet sich in xref:isyfact-standards-doku:isy-security:konzept/master.adoc#inhalt[Konzept des Querschnitt-Bausteins Security].
```

kollmana replied:
Übernommen

huebnercl reviewed (dismissed):
mit einigen Vorschlägen...

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,31 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+NOTE: IAM steht für xref:glossary:literaturextern:inhalt.adoc#litextern-identity_management[Identity and Access Management].
+
+Anwender erhalten Zugriff auf Geschäftsanwendungen innerhalb der IsyFact Anwendungslandschaft nur über Portalanwendungen, Service-Gateways und zukünftig das Authentication-Gateway, die die Außengrenzen der Anwendungslandschaft darstellen.
```
Für Zugriffe innerhalb einer Anwendungs-Landschaft muss das aufrufende System weder eine Authentication-GW noch ein Service-GW aufrufen. Service-GW und Authentication GW braucht man nur wenn eine Anwendung außerhalb der Anwendungslandschaft   auf eine andere Anwendung innerhalb der Anwendungslandschaft zugreifen will

Das Authentication GW sollte man dabei nicht als Außengrenze einer Anwendungslandschaft sehen
Das Authentication GW ist ein (separates) System, das AccessTokens generiert. 
Die AccessTokens werden dann an der Außengrenze des aufgerufenen Anwendungssystems geprüft

kollmana replied:
... Geschäftsanwendungen, die sich innerhalb einer IsyFact Anwendungslandschaft befinden ...

- Das Auth-GW ist Teil der IsyFact Anwendungs**landschaft** und kommuniziert mit externen Anwendungen. 
- Nach Außen fungiert das Auth-GW als Proxy für das IAM, leitet die Daten aber einfach nur weiter und 'versteckt' das IAM nach außen. 
- Die Prüfung das Access-Tokens an der Außengrenze des aufgerufenen Anwendungs**systems** ist die danach kommende Autorisierung der Verwendung dieses Service ("Hat er das dafür nötige Recht?")

kollmana replied:
Anwendungslandschaft = Gesamtheit der Anwendungen/Services/Systeme, die notwendig sind, um eine/mehrere Geschäftsanwendungen (Domänen) eines Kunden (z.B. eine bestimmte Behörde) zu betreiben.

mayrklaus commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,31 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+NOTE: IAM steht für xref:glossary:literaturextern:inhalt.adoc#litextern-identity_management[Identity and Access Management].
+
+Anwender erhalten Zugriff auf Geschäftsanwendungen innerhalb der IsyFact Anwendungslandschaft nur über Portalanwendungen, Service-Gateways und zukünftig das Authentication-Gateway, die die Außengrenzen der Anwendungslandschaft darstellen.
+Dort erfolgt die Authentifizierung des Anwenders (im Bild: schwarzer Schlüssel) mit Hilfe eines zentralen Querschnittservice je Anwendungslandschaft, dem IAM-Service.
```
Was genau meint "dort" ?
Wenn damit die Außengrenze der AWL gemeint ist, dann ist das falsch.
Die Authentifizierung erfolgt **weit vor** der Außengrenze!
Mit erfolgreicher Authentifizierung bekommt das aufrufende System einen Token in die Hand
Und nur mit diesem Token kommt man letztlich in die aufzurufende Anwendung rein.
(Vorausgesetzt das Token ist echt, und die im Token vermerkten Berechtigungen sind hinreichend um den gewünschten Service aufzurufen).

kollmana replied:
Sehe ich anders: 

> Referenzarchitektur - Die fachliche Referenzarchitektur: 
Die Menge aller Anwendungssysteme einer Organisation und deren Nutzungsbeziehungen untereinander bilden eine Anwendungslandschaft. Die fachliche Architektur einer Anwendungslandschaft gibt hier die Strukturierung aus fachlicher Sicht vor und legt fest, wie Anwendungssysteme in die Anwendungslandschaft integriert werden.
Diese fachliche Architektur einer Anwendungslandschaft ist in [Figure 3](https://github.com/IsyFact/isyfact-standards/pull/228#image-GA-AW2) dargestellt. 

Das war die Vorlage für mein Diagramm. Portal, S-GW (u. Auth-GW) sind hier innerhalb der Anwendungslandschaft.

kollmana replied:
Du meintest die Aussengrenze einer einzelnen Geschäftsanwendung/Domäne. 
Dort findet, wie im Absatz darunter beschrieben, die Autorisierung des Zugriffs auf eine Schnittstelle der GA statt.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Included review remarks from C-P.H.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Included review remarks from A.D.:
- reworked diagram
- moved Konzept Security: Prinzipien der Sicherheitsarchitektur to Referenzarchitektur: Zentrale Sicherheitsaspekte in der Referenzarchitektur

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Updated Konzept Sicherheit: Einleitung
- added link to Referenzarchitektur: Zentrale Sicherheitsaspekte
- removed reference to chapter Prinzipien der Sicherheitsarchitektur

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,63 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
 
+[[zentrale-sicherheitsaspekte]]
+=== Zentrale Sicherheitsaspekte in der Referenzarchitektur
+((vorher: Konzept Security: prinzipien-der-sicherheitsarchitektur))
```
Diesen Satz würde ich entfernen.

kollmana replied:
War sowieso nur für uns intern gedacht.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,63 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
 image::blaupausen:referenzarchitektur/CallFAmGuX.png[align="center"]
 
+[[zentrale-sicherheitsaspekte]]
+=== Zentrale Sicherheitsaspekte in der Referenzarchitektur
```
Was hälst du von "Prinzipien der Sicherheitsarchitektur" als Überschrift?

kollmana replied:
Gut - so hiess der Abschnitt im Konzept 'Security'.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,63 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+((vorher: Konzept Security: prinzipien-der-sicherheitsarchitektur))
+
+Die IsyFact-Referenzarchitektur beinhaltet auch Vorgaben für die Sicherheitsarchitektur und für die Prinzipien, nach denen sie aufgebaut ist.
+Eine davon ist die Forderung, dass IsyFact-konforme IT-Systeme gegen unberechtigte Nutzung zu schützen sind:
```
Hier würde ich den Doppelpunkt durch einen Punkt ersetzen und anschließend: "Folgende Prinzipien müssen bei der Erstellung einer Systemlandschaft eingehalten werden."

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,63 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+* Wird zur Ausführung des Aufrufs ein Service einer anderen Geschäftsanwendung / Domäne benötigt, erfolgt dessen Aufruf im Namen und mit der Autorisierung des ursprünglichen Aufrufers / Benutzers.
+* Die feingranulare Autorisierung geschieht an den Systemgrenzen / externen Schnittstellen der aufgerufenen Geschäftsanwendung.
+
+.Integration von IAM in eine Anwendungslandschaft
```
Statt Anwendungslandschaft müssen wir "Systemlandschaft" schreiben, da wir in der software-technischen Sicht unterwegs sind.

kollmana replied:
Innerhalb des Abschnitts ersetzt

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl.dn.svg:
Könntest du das Portal bitte noch breiter machen, sodass rechts und links die Lücken kleiner werden? Gerne auch die beiden Gateways, sodass der Umbruch entfällt.

kollmana replied:
Dann sind die Pfeile zu und von diesen Kästchen nicht mehr mittig bzw. symmetrisch dazu.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl.dn.svg:
Ich würde zudem noch je Zugangsweg 2 Sätze in den Text z.B. unterhalb der Abbildung:

Portal:
Interne und externe Anwender rufen über das Portal die Geschäftsanwendung auf und werden dabei durch das Portal an den IAM-Service weitergeleitet, um die Authentifizierung durchzuführen. Erst nach erfolgreicher Authentifizierung leitet das Portal den Anwender zusammen mit den Autorisierungsinformationen an die Geschäftsanwendung weiter. 

Service-Gateway:
Das Service-Gateway kann die Authentifizierung von externen Anwendungen übernehmen. Dazu leitet es die von der externen Anwendung erhaltenen Authentifizierungsdaten an den IAM-Service weiter. Nach erfolgreicher Authentifizierung leitet das Service-Gateway die Anfrage zusammen mit den Autorisierunginformationen an die Geschäftsanwendung weiter. Zukünftig soll das Service-Gateway keine Authentifizierungsanfragen mehr vearbeiten. Stattdessen soll das Authentication-Gateway verwendet werden.

Authentication-Gateway:
Das Service-Gateway übernimmt die Authentifizierung von externen Anwendungen. Dazu leitet es die von der externen Anwendung erhaltene Authentifizierungsanfrage an den IAM-Service weiter. Nach erfolgreicher Authentifizierung gibt das Authentication-Gateway die Autorisierunginformationen an die externe Anwendung urück. Die externe Anwendung sendet die erhaltenen Autorisierunginformationen gemeinsam mit der fachlichen Anfrage über das Service-Gateway an die Geschäftsanwendung.

Anwendungen innerhalb der Anwendungslandschaft:
Anwendungen innerhalb der Anwendungslandschaft authentifizieren sich mit ihren Zugangsdaten direkt am IAM-Service und nutzen die erhaltenen Autorisierungsinformationen zum Aufruf anderer Geschäfts- und Querschnittsanwendungen.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,63 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+NOTE: IAM steht für xref:glossary:literaturextern:inhalt.adoc#litextern-identity_management[Identity and Access Management].
+
+Das IAM-System verwaltet *Benutzer* (menschliche Anwender und technische Systeme) und ihre *Rollen* zentral für alle Komponenten der Anwendungs- und Systemlandschaft.
```
Bitte IAM-System in IAM-Service wandeln.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Included review remarks from A.Daun
- refined diagram
- textual explanation of components

kollmana committed:
Merge remote-tracking branch 'origin/master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

refinements

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,84 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+
+Die IsyFact-Referenzarchitektur beinhaltet auch Vorgaben für die Sicherheitsarchitektur und für die Prinzipien, nach denen sie aufgebaut ist.
+Eine davon ist die Forderung, dass IsyFact-konforme IT-Systeme gegen unberechtigte Nutzung zu schützen sind.
+Folgende Prinzipien müssen bei der Erstellung einer Systemlandschaft eingehalten werden:
```
Folgende Prinzipien der Sicherheit liegen der Erstellung einer Systemlandschaft zu Grunde:

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,84 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+* Wird zur Ausführung des Aufrufs ein Service einer anderen Geschäftsanwendung / Domäne benötigt, erfolgt dessen Aufruf im Namen und mit der Autorisierung des ursprünglichen Aufrufers.
+* Die feingranulare Autorisierung geschieht an den Systemgrenzen / externen Schnittstellen der aufgerufenen Geschäftsanwendung.
+* Existiert keine Systemlandschaft (d.h. IT-Systeme stehen für sich allein), gilt grundsätzlich, dass die Authentifizierung und grobgranulare Autorisierung in der Nutzungsschicht (GUI, Service oder Batch) des IT-Systems geschehen muss.
+
```
Folgende Abbildung zeigt die Umsetzung der Prinzipen in der Systemlandschaft:

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,84 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+[[prinzipien-der-sicherheitsarchitektur]]
+=== Prinzipien der Sicherheitsarchitektur
+
+Die IsyFact-Referenzarchitektur beinhaltet auch Vorgaben für die Sicherheitsarchitektur und für die Prinzipien, nach denen sie aufgebaut ist.
```
Die IsyFact-Referenzarchitektur beinhaltet auch Vorgaben für die Sicherheitsarchitektur einer Systemlandschaft.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,84 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+* Authentication-Gateway: +
+Das Authentication-Gateway übernimmt die Authentifizierung von externen Anwendungen.
+Dazu leitet es die von der externen Anwendung erhaltene Authentifizierungsanfrage an den IAM-Service weiter.
+Nach erfolgreicher Authentifizierung gibt das Authentication-Gateway die Autorisierungsinformationen an die externe Anwendung zurück. Die externe Anwendung sendet die erhaltenen Autorisierungsinformationen gemeinsam mit der fachlichen Anfrage über das Service-Gateway an die Geschäftsanwendung.
```
Bitte immer nur einen Satz pro Zeile.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/pages/referenzarchitektur/inhalt.adoc:
```diff
@@ -468,6 +468,84 @@ Für derartige Aufrufe gibt es keine Vorgaben: Sie können aus beliebigen Schich
+[id="image-iam-integration-awl",reftext="{figure-caption} {counter:figures}"]
+image::referenzarchitektur/sicherheit_iam_integration_awl.dn.svg[]
+
+* Portal: +
```
Kannst du diese Bullet-Points bitte noch in Fettschrift schreiben?

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/blaupausen/images/referenzarchitektur/sicherheit_iam_integration_awl.dn.svg:
Könntest du den ersten Pfeil beim Webclient (grauer Pfeil) etwas kürzer machen, da dieser in das W reinragt. Bzw. generell die Pfeile etwas kürzer, damit es wieder einheitlich ist.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/konzept/thisdoc.adoc:
```diff
@@ -12,13 +12,12 @@ Die IsyFact richtet sich beim Thema Sicherheit nach behördlichen Standards sowi
 * xref:glossary:literaturextern:inhalt.adoc#litextern-owasp[OWASP Top Ten Project]
 * xref:glossary:literaturextern:inhalt.adoc#litextern-oauth-homepage[OAuth 2.0 Homepage]
 
+Allgemeine xref:blaupausen:referenzarchitektur/master.adoc#prinzipien-der-sicherheitsarchitektur[Zentrale Sicherheitsaspekte in der Referenzarchitektur] werden im Dokument zur xref:blaupausen:referenzarchitektur/master.adoc[Referenzarchitektur] beleuchtet.
```
Hier müsste der Linktext "Prinzipien der Sicherheitsarchitektur" heißen.

dauna-msg commented on isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/konzept/inhalt.adoc:
```diff
@@ -72,7 +19,7 @@ Die *Einfachheit* der Nutzung der Bibliothek `isy-security` wird durch weitgehen
 === Prämissen
 
-Aus den xref:konzept/master.adoc#prinzipien-der-sicherheitsarchitektur[Prinzipien der Sicherheitsarchitektur] leiten sich die folgenden Randbedingungen für die Umsetzung der Berechtigungsprüfung innerhalb eines IT-Systems ab:
+Aus der xref:blaupausen:referenzarchitektur/master.adoc#zentrale-sicherheitsaspekte[Zentralen Sicherheitsaspekten in der Referenzarchitektur] leiten sich die folgenden Randbedingungen für die Umsetzung der Berechtigungsprüfung innerhalb eines IT-Systems ab:
```
Hier müsste der Linktext "Prinzipien der Sicherheitsarchitektur" heißen.

kollmana committed:
docs: ISY-770 [IFS-2009] - reference architecture:  new chapter 'central aspects of security'

Included review remarks from A.Daun
- refined diagram
- wording
- links corrected

kollmana committed:
Merge remote-tracking branch 'origin/master' into feature/ISY-770_IFS-2009_Aenderungen_zentrale_Sicherheitsaspekte_Referenzarchitektur
